src/scenes/Products/Create/CreateProduct.tsx

```diff
@@ -208,10 +209,19 @@ export const CreateProduct = () => {
         <EngagementBreadcrumb data={engagement} />
         <Typography variant="h4">Create Goal</Typography>
       </Breadcrumbs>
-      <Typography variant="h2">
-        {loading ? <Skeleton width="50%" variant="text" /> : 'Create Goal'}
-      </Typography>
+      <Error error={error}>
+        {{
+          NotFound: 'Could not find Engagement',
+          Default: 'Error loading Engagement',
```

Comment:
Can this reference goal creation somehow? Maybe
```suggestion
          NotFound: 'Cannot create goal for this unknown engagement',
          Default: 'Failed to load engagement info needed to create goal',
```